src/Interfaces/SearchHandlerInterface.php

```diff
@@ -19,36 +26,13 @@ public static function getMappings(): array;
      */
     public static function getSettings(): array;
 
-    /**
-     * Get the class this search handler will support
-     *
-     * @return string[] Fully Qualified Class Names that implement the Search Handler interface
-     */
-    public function getHandledClasses(): array;
-
-    /**
-     * Returns the index name that this handler is responsible for.
-     *
-     * @return string
-     */
-    public function getIndexName(): string;
-
-    /**
-     * Returns the identifier used externally for the object.
-     *
-     * @param object $object
-     *
-     * @return mixed|null
-     */
-    public function getSearchId(object $object);
-
     /**
      * Transforms objects supplied into serialized search arrays that
      * should be indexed.
      *
-     * @param mixed $object
+     * @param mixed|null $object
      *
-     * @return mixed[][]
+     * @return mixed[][]|null
      */
-    public function transform($object): ?array;
+    public function transform($object = null): ?array;
```

Comment:
Why should we support nulls?

Reply:
for something like Auditing logs we are not going to be injecting in a doctrine entity, thus nullable. also because I think we discussed that is probably ok to share the transform method for both these cases.

Reply:
Then lets move the transform method to the DoctrineSearchHandler? If the handler has no ability to transform then lets remove that function from the base interface

Reply:
I had it there because the base still needs to be able to allow for implementing class (non entity based) to send documents to index.

Reply:
Sounds to me like the base interface just shouldnt support this functionality if we're going to allow handlers to opt out of transformation.

Alternatively, require the auditing package to implement this method and move its transformation handling into the handler